scenario/consistency_checker.py

```diff
@@ -209,4 +288,10 @@ def check_containers_consistency(
             f"some containers declared in the state are not specified in metadata. That's not possible. "
             f"Missing from metadata: {diff}."
         )
+
+    # guard against duplicate container names
+    names = Counter(state_containers)
+    if dupes := [n for n in names if names[n] > 1]:
+        errors.append(f"Duplicate container name(s): {dupes}.")
+
     return Results(errors, [])
```

Comment:
This reminds me of https://github.com/juju/charm/pull/370

Reply:
apparently it's a long-lived issue :)